### PR TITLE
Sign-In with Ethereum + ERC1271

### PR DESCRIPTION
**Pull request by rkreutz**
Hey there,

This PR adds some structs/classes around Sign-In with Ethereum ([EIP-4361](https://eips.ethereum.org/EIPS/eip-4361)) which is a new standard for trustless sign-ins. This was based on the [TypeScript implementation](https://github.com/spruceid/siwe) of the same feature by the people behind the proposal.

As part of the implementation, I also added [ERC-1271](https://eips.ethereum.org/EIPS/eip-1271) support. To test it I've deployed a couple contracts to Rinkeby, repo is [here](https://github.com/rkreutz/erc1271-contract).

I've added tests and made sure they are also running fine on Linux (since this will be specially useful for Server Side).

Cheers

### CODE REVIEW

**Comment by DarthMike** [comment]
Thanks @rkreutz for the PR. Will take a bit to review as need to read EIP and a bit loaded with other work this week. But definitely good addition to library!

**Comment by dnKaratzas** [comment]
Please check [ERC20Tests.swift](https://github.com/argentlabs/web3.swift/blob/develop/web3sTests/ERC20/ERC20Tests.swift) how to make them to be Tested with `EthereumWebSocketClient` also.

**Comment by rkreutz** [comment]
@dnKaratzas added the tests there

**Comment by dnKaratzas** [comment]
> @dnKaratzas added the tests there

I know but need to rename from `WSS` to `WebSocket`

**Comment by rkreutz** [comment]
@dnKaratzas done

File: web3swift/src/SIWE/SiweVerifier.swift [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    }
+
+    /// Verifies if a given `SiweMessage` was signed by the address in the message.
+    /// - Parameters:
+    ///   - message: the message to be verified
+    ///   - signature: the hexadecimal string of the alleged signature, prefixed with `0x`
+    /// - Returns: a `Bool` indicating if the pair message-signature is verified (whether or not the signature came from the address in the message)
+    /// - Throws: `SiweVerifier.Error` if message is not verifiable;
+    ///           might throw `KeyUtilError` in case recovering the address that signed the message fails
+    open func verify(message: SiweMessage, against signature: String) async throws -> Bool {
```
Comment (DarthMike) [inline web3swift/src/SIWE/SiweVerifier.swift:53]: I think this should be `public` not `open`?

**Review by DarthMike — COMMENTED** [review]
@rkreutz Looks good! Only one small comment, and we can merge it

File: web3swift/src/SIWE/SiweVerifier.swift (hunk 1 above)
Comment (rkreutz) [reply web3swift/src/SIWE/SiweVerifier.swift]: Done [39e0169](https://github.com/argentlabs/web3.swift/pull/246/commits/39e0169c91379493584212e8314f29c75b3f7caa)

**Review by dnKaratzas — COMMENTED** [review]
LGTM!

**Comment by rkreutz** [comment]
@DarthMike I believe this got forgotten at some point? 😄

**Review by DarthMike — APPROVED** [review]

**Comment by DarthMike** [comment]
Thanks @rkreutz, merged

**Pull request merged**